DEV-2320 mh rest api upgrade

lennertvandevelde opened the pull request.

lennertvandevelde committed:
DEV-2320 Switch mediahaven Service 2 mediahaven py

lennertvandevelde committed:
DEV-2320 Cleanup and change Errortypes

lennertvandevelde committed:
DEV-2320 Change tests

lennertvandevelde committed:
DEV-2320 include mediahaven in requirements

spacid commented on app/app.py:
```diff
@@ -16,15 +19,15 @@
 )
 from .services.s3 import S3Client
-from .services.mediahaven_service import MediahavenService, MediaObjectNotFoundException
+# from .services.mediahaven_service import MediahavenService, MediaObjectNotFoundException
```
Best to remove code in comments.

spacid commented on app/app.py:
```diff
@@ -33,27 +36,27 @@ def _get_fragment_metadata(fragment_id: str, mh_client: MediahavenService) -> Di
     try:
-        fragment = mh_client.get_fragment(fragment_id)
-    except MediaObjectNotFoundException as error:
+        fragment = mh_client.records.get(fragment_id)
```
```suggestion
        fragment = mh_client.records.get(fragment_id).single_result
```
This way you don't need to repeat it later on, lines 55-58.

spacid commented on app/app.py:
```diff
@@ -100,14 +103,14 @@ def _generate_vrt_xml(fragment_info: dict, event_timestamp: str) -> str:
     reporting reasons.
 
-    In the case of a valid archived event it will be processed further:
+    In the case of a valid archived event it will bep rocessed further:
```
```suggestion
    In the case of a valid archived event it will be processed further:
```

spacid commented on app/app.py:
```diff
@@ -33,27 +36,27 @@ def _get_fragment_metadata(fragment_id: str, mh_client: MediahavenService) -> Di
-        fragment = mh_client.get_fragment(fragment_id)
-    except MediaObjectNotFoundException as error:
+        fragment = mh_client.records.get(fragment_id)
+    except MediaHavenException as error:
```
This exception can be broader than just a 404.
Or:
 - Expand logging to reflect this.
 - Extra check if the status code is a 404.

spacid commented on app/app.py:
```diff
@@ -184,7 +187,18 @@ def _handle_premis_event(event: PremisEvent, mh_client: MediahavenService):
+    client_secret = config.config["environment"]["mediahaven"]["client_secret"]
+    user = config.config["environment"]["mediahaven"]["username"]
+    password = config.config["environment"]["mediahaven"]["password"]
+    url = config.config["environment"]["mediahaven"]["host"]
```
```suggestion
    mediahaven_config = config.config["environment"]["mediahaven"]
    client_id = mediahaven_config["client_id"]
    client_secret = mediahaven_config["client_secret"]
    user = mediahaven_config["username"]
    password = mediahaven_config["password"]
    url = mediahaven_config["host"]
```

spacid commented on app/app.py:
```diff
@@ -136,8 +139,10 @@ def _handle_premis_event(event: PremisEvent, mh_client: MediahavenService):
-            organisation_name = fragment["Administrative"]["OrganisationName"]
+            fragment = mh_client.records.get(event.fragment_id)
+            # fragment = mh_client.get_fragment(event.fragment_id)
+            # organisation_name = fragment["Administrative"]["OrganisationName"]
```
Best to remove commented code

spacid commented on app/app.py:
```diff
@@ -33,27 +36,27 @@ def _get_fragment_metadata(fragment_id: str, mh_client: MediahavenService) -> Di
+        pid: str = fragment.single_result.Administrative.ExternalId
+        s3_object_key: str = fragment.single_result.Dynamic.s3_object_key
+        s3_bucket: str = fragment.single_result.Dynamic.s3_bucket
+        md5: str = fragment.single_result.Technical.Md5
```
```suggestion
        pid: str = fragment.Administrative.ExternalId
        s3_object_key: str = fragment.Dynamic.s3_object_key
        s3_bucket: str = fragment.Dynamic.s3_bucket
        md5: str = fragment.Technical.Md5
```
See above

lennertvandevelde committed:
DEV-2320 Review changes

lennertvandevelde committed:
DEV-2320 delete deprecated mh service

spacid committed:
DEV-2320 Fix tests

- Patch ROPCGrant mock in tests using a patched client.
- Use MediaHavenSingleObjectJSONMock as return type.

Change requirements.txt:
 - Use latest version of mediahaven Python library.
 - Bump pytest in order to work with Python 3.10.
 - Bump lxml to latest minor.
 - Remove requests as this is packaged with the mediahaven Python library.

spacid committed:
DEV-2320 Format using black

spacid committed:
DEV-2320 remove `.single_result`

This is not needed. Getting an attribute looks through the `single_result`.

Fix respective test.

spacid committed:
DEV-2320 Use Python 3.10

This is the development version used.

Adapt the Dockerfile and Jenkinsfile.

spacid committed:
DEV-2320 Fix python version in Jenkinsfile

Use non-slim version of Debian for running the tests as this has `git`
installed.

spacid committed:
DEV-2320 Fix git issue

Add safe directory

spacid committed:
DEV-2320 Mock config in test

- Mock the config as there is no config.yml file in the repo.
- Remove unused imports.
